torch/_custom_op.py

```diff
@@ -729,7 +733,10 @@ def get_supported_param_types():
         (int, "SymInt", False, True),
         (float, "float", False, True),
         (bool, "bool", False, True),
+        (str, "str", False, False),
         (torch.types.Number, "Scalar", False, False),
+        (torch.dtype, "ScalarType", False, False),
+        (torch.device, "Device", False, False),
```

Comment:
(not a full review yet) the problem is that adding a type T to this list also adds T? and T[] as valid types (see "derived_types"). To avoid the ScalarType problem we should just add another column to this table for if `derived_types` should allow the list variant

Reply:
We might want to also ban str[] and device[] -- I don't see those types in native_functions.yaml and am worried that they might be wrong too in other ways